### PR TITLE
Smtp db

### PR DESCRIPTION
**Pull request by JohnNewman1**

### CODE REVIEW

File: lib/messageDeliveryAgent.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
-
-const Database = require('../models/Database')
 class MessageDeliveryAgent {
-  constructor (database = new Database()) {
+  constructor (database) {
```
Comment (georgesykes86) [inline lib/messageDeliveryAgent.js]: Prefer to call this dbInterface

File: lib/messageDeliveryAgent.js [hunk 2]
```diff
@@ -1,7 +1,5 @@
-
-const Database = require('../models/Database')
 class MessageDeliveryAgent {
-  constructor (database = new Database()) {
```
Comment (charmalt) [inline lib/messageDeliveryAgent.js]: Call dbInterface

File: lib/messageDeliveryAgent.js (hunk 1 above)
Comment (charmalt) [inline lib/messageDeliveryAgent.js]: Call dbInterface

File: models/dbConnection.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const pg = require('pg')
+const env = require('./config')
+
+class DBConnection {
+  constructor (dbClient = new pg.Client(env['test'])) {
```
Comment (charmalt) [inline models/dbConnection.js]: Should be 'development' by default, 'test' only during testing

File: lib/smtpServer.js [hunk 4]
```diff
@@ -2,12 +2,16 @@
...
+const DBConnection = require('../models/dbConnection')
+const SMTPDBInterface = require('../models/smtpDBInterface')
 
 class SMTPServer {
   constructor (port, address) {
     this.port = port || 1337
     this.address = address || '127.0.0.1'
-    this.queue = new MTAQueue()
+    this.databaseconnector = new DBConnection()
+    this.database = new SMTPDBInterface(this.databaseconnector)
```
Comment (georgesykes86) [inline lib/smtpServer.js]: again prefer dbInterface

File: lib/smtpServer.js [hunk 5]
```diff
@@ -2,12 +2,16 @@
...
 const MTAQueue = require('./mtaQueue')
+const DBConnection = require('../models/dbConnection')
+const SMTPDBInterface = require('../models/smtpDBInterface')
 
 class SMTPServer {
   constructor (port, address) {
     this.port = port || 1337
     this.address = address || '127.0.0.1'
-    this.queue = new MTAQueue()
+    this.databaseconnector = new DBConnection()
```
Comment (georgesykes86) [inline lib/smtpServer.js]: Prefer dbConnection

File: models/smtpDBInterface.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+class SMTPDBInterface {
+  constructor (connection) {
+    this.connection = connection
+  }
+
+  async post (mailto, mailfrom, mailbody) {
```
Comment (charmalt) [inline models/smtpDBInterface.js]: great work!

File: models/dbConnection.js (hunk 3 above)
Comment (georgesykes86) [inline models/dbConnection.js]: Agree with Charlene this should default to development

**Comment by nixlim** [comment]
:shipit:

**Pull request merged**